modules/core/src/main/java/io/fluo/webindex/core/models/URL.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2016 Fluo authors (see AUTHORS)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.fluo.webindex.core.models;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class URL implements Serializable {
```

Comment:
I don't have a suggestion for another name... but was thinking this class name is slightly confusing since Java stdlibs have a class w/ same name